src/MainEmpresaProgramadores.java

```diff
@@ -1,103 +1,135 @@
-import java.util.Scanner;
+import programacion.empresaprogramacion.Empleado;
 import programacion.empresaprogramacion.Empresa;
 import programacion.empresaprogramacion.Gerente;
 import programacion.empresaprogramacion.Programador;
+import programacion.empresaprogramacion.excepciones.ParametroInvalidoException;
+import programacion.empresaprogramacion.excepciones.RutaArchivoInvalida;
+
+import java.util.Scanner;
 
 public class MainEmpresaProgramadores {
+  public static Scanner sc = new Scanner(System.in);
 
   public static void main(String[] args) {
 
-    Empresa empresa = new Empresa("Empresa del Juan de la Cierva");
-    int opcion = 0;
-
-    Scanner scanner = new Scanner(System.in);
-
-    referencia = new Programador(parametros);
-    empresa.addEmpleado(referencia);
-
-    referencia = new Programador(parametros);
-    empresa.addEmpleado(referencia);
-
-    referencia = new Gerente(parametros);
-    empresa.addEmpleado(referencia);
-
-    referencia = new Gerente(parametros);
-    empresa.addEmpleado(referencia);
-
-    referencia = new Programador(parametros);
-    empresa.addEmpleado(referencia);
-
-    referencia = new Programador(parametros);
-    empresa.addEmpleado(referencia);
-
-    referencia = new Programador(parametros);
-    empresa.addEmpleado(referencia);
-
-    referencia = new Programador(parametros);
-    empresa.addEmpleado(referencia);
+    Empresa empresa = new Empresa("Tecnologías Gyllenhaal");
+    int opcion;
+
+    Empleado referencia;
+
+    try {
+      referencia = new Programador("50549668S", "Daniel", 1302.93, "Java", true);
+      empresa.addEmpleado(referencia);
+    } catch (ParametroInvalidoException e) {
+      System.out.println(e.getMessage());
```

Comment:
@ianharrisonromero Algo así creo que daría más información:

```suggestion
      System.out.println("Excepción capturada creando Programador: " + e.getMessage());
```

Pero por el formato de los mensajes al capturar una excepción no voy a quitar puntos: quiero que sepáis cómo sacar el mensaje de una excepción y que pongáis el try/catch donde corresponde.

Reply:
Acabo de caer en que, si en el try se puede lanzar la misma excepción desde el constructor de Programador y desde el método `addEmpleado`, no tendríamos forma de saber de dónde viene la excepción.
En un caso real, cada método lanzaría distintas excepciones y/o el mensaje de la excepción sería suficientemente claro para saber qué ha ocurrido y dónde.
Pero esto es demasiado enrevesado para pedir afinar tanto en un examen...